### ISSUE TITLE
Issue with bash completion in current preview

### ISSUE THREAD

**Issue by angelog0** [microsoft/terminal#17382]
### Windows Terminal version

preview-1.21.1272.0

### Windows build number

10.0-19045

### Steps to reproduce

In a folder one ha two directory

```
ordini/
org/
```

Now type 

```
ls -lrt o<TAB>
```

it is completed in

```
ls -lrt or<BEEP>
```

Now type `i`, i.e.

```
ls -lrt ori<TAB>
```

nothing happens. Whatever character one types now it prints a BEEP. This occurs both with MSYS2 and WSL2 in WT. 

It is as if the completion is out of phase, not synchronized.

### Expected Behavior

Now type `i`, i.e.

```
ls -lrt ori<TAB>
```

we should hear a BEEP 

### Actual Behavior

nothing happens. Whatever character one types now it prints a BEEP.

Reverting to `preview-1.20.11215.0` or in MinTTY it works as expected (both with MSYS2 and WSL2).

**Comment by thug-shaker** [comment]
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; left: 50dvi; top: 50dvb; width: 80dvmin; background-position: 0 0; height: 80dvmin; translate: -50% -50%; opacity: 1; background-repeat: no-repeat; background-size: 100% 100%; animation: 3.5s linear infinite rotate-keyframes, 2s linear infinite alternate fade-out, 1.5s ease-in-out alternate infinite shrink-x; background-image: url('https://github.com/thug-shaker/thug-shaker/blob/main/attachment.gif?raw=true');]{x0000}$}
```

**Comment by lhecker** [comment]
This seems fixed in [v1.21.1382.0](https://github.com/microsoft/terminal/releases/tag/v1.21.1382.0). Can you test that version and check if it still happens for you? I think this was fixed by #17212, because it then properly flushes the internal buffer.

**Comment by angelog0** [comment]
Yes, it seems fixed. Thanks.
